### PR TITLE
Intrinio source 

### PR DESCRIPTION
**Pull request by jmaslek**
This PR was my project from the holiday.  It does the following

- adds Intrinio as a `stocks/load` source
- adds intrinio as an sdk option load
- Adds calculations for delta neutral/gamma neutral/gamma max via the sdk

Note that the options menu needs some refactoring and I was going to touch it here, but I wanted to break into smaller PRs if I did it.

### CODE REVIEW

File: openbb_terminal/stocks/options/op_helpers.py [hunk 1]
```diff
@@ -398,3 +408,171 @@ def Vomma(self, change):
...
+        df.implied_volatility.to_numpy(),
+        df.dte.to_numpy(),
+        df.type.to_numpy(),
+        rf_rate,
+    )
+    return np.abs(np.sum(df["new_delta"].to_numpy() * df["open_interest"].to_numpy()))
+
+
+@log_start_end(log=logger)
+def get_market_gamma(
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/op_helpers.py:527]: This function is not being used anywhere?

File: openbb_terminal/stocks/options/options_controller.py [hunk 2]
```diff
@@ -585,6 +588,8 @@ def call_load(self, other_args: List[str]):
                 self.expiry_dates = yfinance_model.option_expirations(self.ticker)
             elif ns_parser.source == "Nasdaq":
                 self.expiry_dates = nasdaq_model.get_expirations(self.ticker)
+            elif ns_parser.source == "Intrinio":
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_controller.py:591]: Probably better after the refactor

File: openbb_terminal/stocks/options/options_controller.py [hunk 3]
```diff
@@ -665,6 +670,11 @@ def call_exp(self, other_args: List[str]):
                             self.ticker, self.selected_date
                         )
                         self.chain = op_helpers.Chain(df, self.source)
+                    elif self.source == "Intrinio":
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_controller.py:673]: Probably better after the refactor

File: openbb_terminal/stocks/options/options_controller.py [hunk 4]
```diff
@@ -1463,3 +1473,31 @@ def call_hedge(self, _):
...
+            help="Date to get option chain for.  Format YYYY-MM-DD",
+        )
+
+        ns_parser = self.parse_known_args_and_warn(parser, other_args)
+
+        if ns_parser:
+            if not self.ticker:
+                console.print("No ticker loaded. First use `load <ticker>`")
+                return
+            # TODO: Make all models return self.data
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_controller.py:1500]: Can probably be done differently after the refactor --> we'll be loading always the full option chain

File: openbb_terminal/stocks/options/options_sdk_helper.py [hunk 5]
```diff
@@ -61,6 +65,14 @@ def get_full_option_chain(
         if expiration:
             return nasdaq_model.get_chain_given_expiration(symbol, expiration)
         return nasdaq_model.get_full_option_chain(symbol)
+    if source == "Intrinio":
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_sdk_helper.py:68]: Prob the refactor will conflict this (my fault) I can help resolve it

File: openbb_terminal/stocks/options/options_sdk_helper.py [hunk 6]
```diff
@@ -139,3 +153,34 @@ def hist(
...
+    date : str
+        Date to get delta neutral price for
+    x0 : float, optional
+        Optional initial guess for solver, defaults to close price of that day
+
+    Returns
+    -------
+    float
+        Delta neutral price
+    """
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_sdk_helper.py:174]: We might want to add an example for the sdk

File: openbb_terminal/miscellaneous/data_sources_default.json [hunk 7]
```diff
@@ -6,7 +6,7 @@
     "candle": [],
     "codes": ["Polygon"],
     "news": ["Feedparser", "NewsApi"],
-    "load": ["YahooFinance", "IEXCloud", "AlphaVantage", "Polygon", "EODHD"],
+    "load": ["YahooFinance", "IEXCloud", "AlphaVantage", "Polygon", "EODHD", "Intrinio"],
```
Comment (hjoaquim) [inline openbb_terminal/miscellaneous/data_sources_default.json:9]: Not sure why, but I needed to add the source manually on `OpenBBUserData/data_sources_default.json` in order to use it

**Review by hjoaquim — COMMENTED** [review]
Added a couple of comments. Overall, this seems great!

Two small things that I think we should add:

1. Add to the keys menu
   1.1 In line with that, we'll probably want to add the `check_api_key` decorator in the model methods
2. Add some integration tests
   2.1 stocks menu
   2.2 options menu

Also, should we wait for the #3665 PR to be merged? I expect a couple of conflicts

**Comment by jmaslek** [comment]
> Also, should we wait for the #3665 PR to be merged? I expect a couple of conflicts

Yeah I am going to wait for that one.

**Pull request closed without merging**